version: "3.9"

services:
  localstack:
    container_name: localstack
    image: fiapsoat2grupo13/localstack-resources:latest
    platform: linux/amd64
    ports:
      - "127.0.0.1:4566:4566"            # LocalStack Gateway
      - "127.0.0.1:4510-4559:4510-4559"  # external services port range
    environment:
      # LocalStack configuration: https://docs.localstack.cloud/references/configuration/
      - DEBUG=${DEBUG:-0}
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
    networks:
      - tech-challenge-network-local

  mongo:
    container_name: mongodb
    image: mongo
    restart: always
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: admin
    ports:
      - "27017:27017"
    networks:
      - tech-challenge-network-local

  redis:
    container_name: redis
    image: library/redis:6-alpine
    platform: linux/amd64
    ports:
      - "6380:6379"
    healthcheck:
      test: [ "CMD", "redis-cli", "-p", "6379", "ping" ]
      interval: 5s
      timeout: 5s
      retries: 3
    networks:
      - tech-challenge-network-local

networks:
  tech-challenge-network-local:
    driver: bridge
